lib/http/request/writer.rb

```diff
@@ -47,7 +47,11 @@ def connect_through_proxy
       # Adds the headers to the header array for the given request body we are working
       # with
       def add_body_type_headers
-        return if @headers[Headers::CONTENT_LENGTH] || chunked?
+        return if @headers[Headers::CONTENT_LENGTH] || chunked? || (
+          @body.source.nil? && %w[GET HEAD DELETE CONNECT].any? do |method|
+            @request_header[0].start_with?("#{method} ")
+          end
+        )
```

Comment:
I guess I'm too late, but I think this would look easier to read:

``` ruby
return if @headers[Headers::CONTENT_LENGTH] || chunked?

content_length = @body.size
return if content_length.zero? && (
  %w[GET HEAD DELETE CONNECT].any? do |method|
    @request_header[0].start_with?("#{method} ")
  end
)

@request_header << "#{Headers::CONTENT_LENGTH}: #{@body.size}"
```

Also, seem like it will be more efficient to pass request object into writer for this o-O

Reply:
@ixti yeah, I thought the logic here could be better, although I didn't consider that and it seems nice.

It hasn't been released yet so....

Reply:
Checking @body.size is not correct, it won't send a Content-Length when the body is an empty string. The other changes seem fine though :)